Dockerfile

```diff
@@ -95,13 +93,21 @@ RUN if test -e modules/reana-commons; then \
       fi \
     fi
 
+# hadolint ignore=DL3013
+RUN if [ "${DEBUG}" -gt 0 ]; then pip install --no-cache-dir -e ".[debug,xrootd]"; else pip install --no-cache-dir ".[xrootd]"; fi;
+
```

Comment:
Why do we need to move this after the installation of reana-commons?

Reply:
When building locally in debug mode and installing a version of reana-commons that is only available locally (not released on PyPi yet), this step was failing because it could not find the required reana-commons version.
Moving it after the installation of reana-commons solves the issue.

Reply:
Any reasons why you needed to change the version of r-commons locally? Just to understand what is the use case

Reply:
It's just usually what I do when updating r-commons and some other component at the same time, especially because I can quickly check if the changes were propagated correctly by just looking at the installed r-commons version

Reply:
FWIW it would be nice to keep the same order for all the REANA components, i.e. either make this change  everywhere, or just keep the original order here too.

Personally when I was testing your branch, I just downgraded the `setup.py` line to require 0.9.5 by a local edit, and tested it that way. 

IMO the old order was acceptable, because we usually only bump r-commons version when the new r-commons is released on PyPI, so the branch remains in a "easily testable" locally for a long time, so the above `setup.py` micro-editing is not even needed.

That said, we could also change the order if it make things easier -- and there should be no danger of overriding some dependencies in that way -- but then let's do it for all the components systematically, for the overall consistency, and in a nicely separate commit,  for better housekeeping.